Add library go tools

juanvallejo opened the pull request:
cc @tnozicka @soltysh @deads2k

soltysh commented on Makefile:
```diff
@@ -1,7 +1,37 @@
+$(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
+
+clean:
+	$(RM) ./must-gather
```
That needs to be `openshift-dev-helpers` and `openshift-must-gather`

tnozicka commented on Makefile:
```diff
@@ -1,7 +1,33 @@
+# and also hooked into {update,verify}-generated for broader integration.
+$(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
+
+clean:
```
drop clean, you'll get it with the multiple binary patch automatically

tnozicka commented on glide.yaml:
```diff
@@ -12,3 +12,6 @@ import:
   version: master
+- package: github.com/openshift/library-go
+  repo:    https://github.com/soltysh/library-go
+  version: rebase-1.12.2
```
setup your editor to always put "\n" at the end of the file :)

juanvallejo replied:
:) will do

tnozicka commented on Makefile:
```diff
@@ -1,7 +1,33 @@
+# $5 - output
+# It will generate targets {update,verify}-bindata-$(1) logically grouping them in unsuffixed versions of these targets
+# and also hooked into {update,verify}-generated for broader integration.
+$(call add-bindata,v3.11.0,./bindata/v3.11.0/...,bindata,v311_00_assets,pkg/operator/v311_00_assets/bindata.go)
```
I don't think this repo uses bindata

juanvallejo replied:
Thanks, removed

tnozicka commented:
@juanvallejo when you are making a dependency bump the glide files should be in a separate commit and then the vendor update goes into commit named `bump(*)`. Makes it easier to review - you check the glide files and CI checks that the vendor content matches what you specified in the glide. You optionally review the bump content based on circumstances.

juanvallejo commented:
@tnozicka thanks. Went ahead and split the bump commit into two separate ones. Renamed commit updating vendored deps to `bump(*)`. Additional comments also addressed

tnozicka commented on Dockerfile:
```diff
@@ -0,0 +1,10 @@
+RUN go build -ldflags "-X $GO_PACKAGE/pkg/version.versionFromGit=$(git describe --long --tags --abbrev=7 --match 'v[0-9]*')" ./cmd/openshift-must-gather
+
+FROM registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+RUN mkdir -p /usr/share/bootkube/manifests
```
why we need this line?

juanvallejo replied:
Sorry, leftover from copying the existing Makefile from another repo. Removed

juanvallejo commented:
> should we wait for rebase and vendoring library-go from the proper branch before merging?

@soltysh I'll defer this to you. I am currently using the 1.12 library-go branch from your repo, as discussed.

soltysh commented:
> > should we wait for rebase and vendoring library-go from the proper branch before merging?
> 
> @soltysh I'll defer this to you. I am currently using the 1.12 library-go branch from your repo, as discussed.

Yeah it's fine, although we still need to wait for https://github.com/openshift/library-go/pull/200 to land before merging this.

tnozicka commented:
I am fine using @soltysh fork if it's based on merged PR

juanvallejo commented:
> I am fine using @soltysh fork if it's based on merged PR

@tnozicka it's based on https://github.com/soltysh/library-go/tree/rebase-1.12.2

soltysh commented:
@juanvallejo you can rebase on top of latest code, I've already picked Tomes's code.

juanvallejo commented:
@soltysh 
> @juanvallejo you can rebase on top of latest code, I've already picked Tomes's code.

Thanks, done

soltysh commented:
/retest

soltysh commented:
/retest

juanvallejo commented:
/test e2e-aws-operator
